keras/regularizers.py

```diff
@@ -23,4 +23,23 @@ def l1l2wrap(g, p):
     return l1l2wrap
 
 def identity(g, p):
-    return g
\ No newline at end of file
+    return g
+
+
+def activity_l1(l=.01):
+    # activity dependent l1 norm
+    def l1wrap(layer):
+        # needs to be wrapped twice because input is not present during instantiation
+        def l1wrap_wrap():
```

Comment:
Probably the way to go would be to make regularizers a class. We could have a class that includes both the old (gradient updating) and new (cost updating) functions. They would then implement either get_cost_update or get_gradient_update.

How do you feel about that? I am a bit busy right now but I could get to it around the middle of the week.